iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -106,6 +113,26 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             }
         }
 
+        private void checkTimeout(Object _)
+        {
+            if (_twinResponseTimeouts.Any())
+            {
+                var currentDateTime = DateTime.UtcNow;
+                TimeSpan difference;
+                foreach (KeyValuePair<string, DateTime> entry in _twinResponseTimeouts)
+                {
+                    difference = currentDateTime - entry.Value;
+                    if (difference >= _twinResponseTimeout)
+                    {
+                        if (_twinResponseCompletions.ContainsKey(entry.Key))
```

Comment:
Do we even need a check to ContainsKey? TryRemove won't fault if it isn't there.